plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -304,6 +318,108 @@ private static String quotedName(String name)
         return '"' + name.replace("\"", "\"\"") + '"';
     }
 
+    public static boolean canEnforceColumnConstraintInAllSpecs(TypeOperators typeOperators, Table table, IcebergColumnHandle columnHandle, Domain domain)
+    {
+        return table.specs().values().stream()
+                .allMatch(spec -> canEnforceConstraintWithinPartitioningSpec(typeOperators, spec, columnHandle, domain));
+    }
+
+    private static boolean canEnforceConstraintWithinPartitioningSpec(TypeOperators typeOperators, PartitionSpec spec, IcebergColumnHandle column, Domain domain)
+    {
+        for (PartitionField field : spec.fields()) {
+            if (field.transform().toString().equals("void")) {
+                // Useless for filtering.
+                continue;
+            }
+            if (field.sourceId() != column.getId()) {
+                // Not related to each other.
+                continue;
+            }
+            if (field.transform().isIdentity()) {
+                // A predicate on an identity partitioning column can always be enforced.
+                return true;
+            }
+
+            ColumnTransform transform = PartitionTransforms.getColumnTransform(field, column.getType());
+            if (transform.isMayReturnNullForNonNull()) {
+                // Partitioning transform must return NULL for NULL input.
+                // Below we assume it never returns NULL for non-NULL input,
+                // so NULL values and non-NULL values are always segregated.
+                // In practice, this condition matches the void transform only,
+                // which isn't useful for filtering anyway.
+                continue;
+            }
+            ValueSet valueSet = domain.getValues();
+
+            boolean canEnforce = valueSet.getValuesProcessor().transform(
+                    ranges -> {
+                        MethodHandle targetTypeEqualOperator = typeOperators.getEqualOperator(
+                                transform.getType(), InvocationConvention.simpleConvention(FAIL_ON_NULL, NEVER_NULL, NEVER_NULL));
+                        for (Range range : ranges.getOrderedRanges()) {
+                            if (!canEnforceRangeWithPartitioningField(field, transform, range, targetTypeEqualOperator)) {
+                                return false;
+                            }
+                        }
+                        return true;
+                    },
+                    discreteValues -> false,
+                    allOrNone -> true);
+            if (canEnforce) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean canEnforceRangeWithPartitioningField(PartitionField field, ColumnTransform transform, Range range, MethodHandle targetTypeEqualOperator)
+    {
+        if (!transform.isMonotonic()) {
```

Comment:
At the time of this writing, the PR has only a test applying for `bucket(...)` transformation.

Do you plan to add more tests to add coverage for the rest of the code of this method as well?

Reply:
that's why it's draft